### PR TITLE
[3.11] GlusterFS: Upgrade playbook fixes

### PR DESCRIPTION
**Pull request by jarrpa**
This PR fixes a few small bugs found in testing the GlusterFS upgrade playbook.

Resolves:
- https://bugzilla.redhat.com/show_bug.cgi?id=1660439
- https://bugzilla.redhat.com/show_bug.cgi?id=1694937
- https://bugzilla.redhat.com/show_bug.cgi?id=1694960

### CODE REVIEW

**Comment by jarrpa** [comment]
@michaelgugino Who should have a look at the changes to openshift_node?

**Comment by jarrpa** [comment]
@vrutkovs Is the e2e-gluster test gone?

File: playbooks/openshift-glusterfs/private/setup_nodes.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@
   - import_role:
       name: openshift_node
       tasks_from: glusterfs.yml
+  - import_tasks: roles/openshift_node/tasks/storage_plugins/iscsi.yml
```
Comment (vrutkovs) [inline playbooks/openshift-glusterfs/private/setup_nodes.yml]:
```suggestion
  - import_role: openshift_node
    tasks_from: storage_plugins/iscsi.yml
```

File: playbooks/openshift-glusterfs/private/upgrade.yml [hunk 2]
```diff
@@ -1,8 +1,36 @@
 ---
+- name: GlusterFS Upgrade Checkpoint Start
+  hosts: all
```
Comment (vrutkovs) [inline playbooks/openshift-glusterfs/private/upgrade.yml]:
```suggestion
  hosts: OSEv3
```

File: playbooks/openshift-glusterfs/private/upgrade.yml [hunk 3]
```diff
@@ -1,8 +1,36 @@
...
   tasks:
   - name: Run glusterfs upgrade role
     import_role:
       name: openshift_storage_glusterfs
       tasks_from: upgrade.yml
+
+- import_playbook: setup_nodes.yml
+
+- name: GlusterFS Upgrade Checkpoint End
+  hosts: all
```
Comment (vrutkovs) [inline playbooks/openshift-glusterfs/private/upgrade.yml]:
```suggestion
  hosts: OSEv3
```

File: playbooks/openshift-glusterfs/private/setup_nodes.yml (hunk 1 above)
Comment (jarrpa) [reply playbooks/openshift-glusterfs/private/setup_nodes.yml]: I tried this first, but travis complained. Some linter couldn't find `file 'iscsi.yml'`.

File: playbooks/openshift-glusterfs/private/upgrade.yml (hunk 2 above)
Comment (jarrpa) [reply playbooks/openshift-glusterfs/private/upgrade.yml]: I'm just following the patterns I'm seeing in other files. Is this a new standard?

**Comment by jarrpa** [comment]
@vrutkovs Ping?

File: playbooks/openshift-glusterfs/private/upgrade.yml (hunk 2 above)
Comment (vrutkovs) [reply playbooks/openshift-glusterfs/private/upgrade.yml]: This has always been a standard Gluster has been ignoring :)

The inventory may contain hosts, which are not part of OSE, so we should use OSEv3 group to minimize changes

File: playbooks/openshift-glusterfs/private/setup_nodes.yml (hunk 1 above)
Comment (vrutkovs) [reply playbooks/openshift-glusterfs/private/setup_nodes.yml]: Ah, okay, lets stick to this then

File: playbooks/openshift-glusterfs/private/upgrade.yml (hunk 2 above)
Comment (jarrpa) [reply playbooks/openshift-glusterfs/private/upgrade.yml]: Gotcha.

**Comment by jarrpa** [comment]
@vrutkovs Groups changed.

File: playbooks/openshift-glusterfs/private/setup_nodes.yml (hunk 1 above)
Comment (mtnbikenc) [reply playbooks/openshift-glusterfs/private/setup_nodes.yml]: The problem is that `tasks_from:` cannot reference files in subdirectories under `tasks\`.  There is an Ansible bug on this somewhere.

**Comment by jarrpa** [comment]
@vrutkovs ping?

File: playbooks/openshift-glusterfs/private/upgrade.yml [hunk 4]
```diff
@@ -1,8 +1,36 @@
 ---
+- name: GlusterFS Upgrade Checkpoint Start
+  hosts: OSEv3
```
Comment (sdodson) [inline playbooks/openshift-glusterfs/private/upgrade.yml]: Sorry, we should've provided more guidance, can you use oo_all_hosts group instead. Here and everywhere else you're using OSEv3?

Comment (jarrpa) [reply playbooks/openshift-glusterfs/private/upgrade.yml]: Done.

**Comment by kristvanbesien** [comment]
This breaks the install of glusterfs on atomic. 
The import of the iscsi.yml tasks should be conditional on the node not being atomic. Like this:

 ```
 - import_tasks: roles/openshift_node/tasks/storage_plugins/iscsi.yml
    when:
      - "'iscsi' in osn_storage_plugin_deps"
      - not openshift_is_atomic | bool

```

**Comment by jarrpa** [comment]
@kristvanbesien Addressed in #11557.

**Pull request merged**